ext/standard/streamsfuncs.c

```diff
@@ -616,93 +616,119 @@ PHP_FUNCTION(stream_get_wrappers)
 }
 /* }}} */
 
-/* {{{ stream_select related functions */
-static int stream_array_to_fd_set(zval *stream_array, fd_set *fds, php_socket_t *max_fd)
-{
-	zval *elem;
-	php_stream *stream;
-	int cnt = 0;
+/* {{{ fd_bigset related macros and typedefs to remove FD_SETSIZE limitation in stream_select */
+typedef struct {
+#ifdef __USE_XOPEN
+    char *fds_bits;
+#else
+    char *__fds_bits;
+#endif
+    size_t size;
+} fd_bigset;
+
+#ifdef __USE_XOPEN
+#define FD_BIGSET_ZERO(set, num_fds) do { \
+    (set)->size = (num_fds + 7) / 8; \
+    (set)->fds_bits = (char *)ecalloc((set)->size, sizeof(char)); \
+} while (0)
+#define FD_BIGSET_SET(fd, set) ((set)->fds_bits[(fd) / 8] |= (1 << ((fd) % 8)))
```

Comment:
The `PHP_SAFE_FD_*` macros check the set size to avoid out of bound writes. Looking at the changes, I believe that we still need these checks in `FD_BIGSET_*`.

Reply:
I understand the need for PHP_SAFE_FD_* macros when you have a fixed size for the fd_set, but fd_bigset is created so that it can fit the largest file descriptor id in the provided arrays. Not sure how that macro could be created so that it knows in advance how big the fd_bigset is.

Reply:
@arnaud-lb I have updated the macros to include a test to check if they are attempting to write outside of the bounds of the fd_bigset->size. Pushing when testing is complete locally.